src/v/redpanda/application.cc

```diff
@@ -797,6 +797,11 @@ void application::wire_up_redpanda_services() {
       fetch_session_cache,
       config::shard_local_cfg().fetch_session_eviction_timeout_ms())
       .get();
+    _deferred.emplace_back([this] {
+        partition_manager
+          .invoke_on_all(&cluster::partition_manager::shutdown_all)
+          .get();
+    });
```

Comment:
If it's not possible to shuffle construction of `_rpc` before `partition_manager`, or attach `shutdown_all` to `stop`, then it probably deserves a comment about lifetimes.

Reply:
I agree. In cases like this the commit message can (and should) provide lots of supplemental information, but a short note (which can even point at the commit message for more information) should also be left at the call site where ordering is important.

Reply:
I've added a comment explaining why is it not possible to shuffle the sequence of staring partition manager vs kafka server. And also a comment explaining why this is needed.